### PR TITLE
feat: reboot via kexec

### PR DESCRIPTION
**Pull request by smira**
This should save a lot of time on BIOS/POST time with bare metal
hardware.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

**Comment by smira** [comment]
requires https://github.com/talos-systems/pkgs/pull/334

File: internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go [hunk 1]
```diff
@@ -1664,3 +1670,77 @@ func ActivateLogicalVolumes(seq runtime.Sequence, data interface{}) (runtime.Tas
...
+		defer kernel.Close() //nolint:errcheck
+
+		initrd, err := os.Open(initrdPath)
+		if err != nil {
+			return err
+		}
+
+		defer initrd.Close() //nolint:errcheck
+
+		parameters, err := ioutil.ReadFile("/proc/cmdline")
```
Comment (smira) [inline internal/app/machined/pkg/runtime/v1alpha1/v1alpha1_sequencer_tasks.go:1718]: this is probably wrong way, as it reads current boot args vs. what is in GRUB

**Comment by smira** [comment]
Q: do we need a kernel boot arg to disable kexec? should kexec be disabled or enabled by default?

**Review by andrewrynhard — APPROVED** [review]
Can't wait to try this!

**Pull request merged**